go/mysql/json/marshal.go

```diff
@@ -36,7 +36,7 @@ func (v *Value) MarshalSQLTo(dst []byte) []byte {
 }
 
 func (v *Value) marshalSQLInternal(top bool, dst []byte) []byte {
-	switch v.t {
+	switch v.Type() {
```

Comment:
This ensures we never see a `typeRawString` but always a properly decoded `TypeString`.